backend/services/product/internal/product/product.go

```diff
@@ -0,0 +1,116 @@
+package product
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/manaaan/ekolivs-oms/pkg/env"
+	"github.com/manaaan/ekolivs-oms/pkg/zettle"
+	"github.com/manaaan/ekolivs-oms/product/api"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type Service struct {
+	zettleService *zettle.Service
+	storeService  *Store
+}
+
+func New() (*Service, error) {
+	zettleService, err := zettle.New(zettle.ServiceNewParams{
+		ClientId: env.Required("ZETTLE_ORG_UUID"),
+		ApiKey:   env.Required("ZETTLE_API_KEY"),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &Service{
+		zettleService: zettleService,
+		storeService:  &Store{},
+	}, nil
+}
+
+func (s Service) GetProducts() ([]*api.Product, error) {
+	zettleProducts, err := s.zettleService.GetProducts()
+	if err != nil {
+		return nil, err
+	}
+
+	storeProducts, err := s.storeService.GetProducts()
+	fmt.Println(storeProducts)
+
+	products := []*api.Product{}
+	for _, zettleProduct := range zettleProducts {
+		for _, variant := range zettleProduct.Variants {
+			var imageUrl *string
+			if zettleProduct.Presentation != nil {
+				imageUrl = zettleProduct.Presentation.ImageUrl
+			}
+
+			products = append(products, &api.Product{
+				Id:            variant.Uuid.String(),
+				Name:          zettleProduct.Name,
+				Sku:           variant.Sku,
+				Barcode:       variant.Barcode,
+				Price:         convertToPrice(variant.Price),
+				CostPrice:     convertToPrice(variant.CostPrice),
+				ImageUrl:      imageUrl,
+				VatPercentage: zettleProduct.VatPercentage,
+				Status:        api.Status_ACTIVE,
+				UnitType:      convertToUnitType(zettleProduct.UnitName),
+				CreatedAt:     convertToTimestamp(zettleProduct.Created),
+				UpdatedAt:     convertToTimestamp(zettleProduct.Updated),
```

Comment:
Now that I see it in the response, we should probably adjust this response to a ISO8601 response for frontend usage and simpler readability of the data.
What are your thoughts?

Reply:
Do I get this right: You want to pass ISO8601 string dates to the client?
If that's right, we don't need to convert them, we can just pass the strings we get from Zettle, they are already in this format.
At the moment I don't see an advantage in converting the date string to a timestamp, as we don't manipulate it. Having it as a ISO8601 string value makes it easier to debug visually and to create Dates in JS (very specific to the client). We can just pass the string value to the Date constructor. With the `timestamppb.Timestamp` we can only access the seconds. We would need to multiply them to get milliseconds and pass it to the Date constructor.
I see three options here:
1. Use a time struct everywhere.
2. Use an ISO8601 everywhere. Functions that modify it should also return a ISO8601 string.
3. Only use a time struct where we need to modify, without the restriction from 2.

Maybe I get it totally wrong. Let's discuss it next time!

Reply:
We decided to keep ISO8601 strings.